.github/workflows/deploy.yml

```diff
@@ -41,7 +41,7 @@ jobs:
 
       - name: Publish PR change to staging site
         uses: peaceiris/actions-gh-pages@v3
-        if: github.event_name == 'pull_request'
+        if: github.event_name != 'pull_request'
```

Comment:
Another way could be that we have two deploy yml for different branches (one behaviour for main/prod branch versus others)

Reply:
I tried both == and != and this doesn't work for staging. I must have missed some other setting somewhere.